usr/src/cmd/iscsid/iscsi-initiator

```diff
@@ -170,6 +171,23 @@ mount_iscsi() {
 }
 
 umount_iscsi () {
+	# In case any non-"zones" zpools reside on iscsi drives, export
+	# all of them now so that the system won't hang later in the shutdown
+	# sequence when those zpools need to be unmounted.
+	zpools=$(zpool list -Ho name)
+	if [ -n $zpools ]; then
+		for pool in $zpools; do
+			if [ "$pool" == "zones" ]; then
+				continue
+			fi
+			zpool export $pool
```

Comment:
Can we use & here to export them all at once?
Then "wait" after the loop. We should get an error messages in the log, and I don't think that in general an export failure should stop shut down (consider a smartos user who is usign a pool for some other situation)

Reply:
Not sure if you decided this wasn't worth doing, or you decided against it?

Reply:
Sorry, I thought I had already responded to this comment. Yes, I decided it wasn't worth the extra complexity of doing that since we're usually only going to have 0, or maybe 1-2, zpools over iscsi. However, I did change the code to not fail the shutdown if an export were to fail.